shell/platform/darwin/ios/framework/Headers/FlutterCodecs.h

```diff
@@ -0,0 +1,83 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FLUTTERCODECS_H_
+#define FLUTTER_FLUTTERCODECS_H_
+
+#import <Foundation/Foundation.h>
+#include "FlutterMacros.h"
+
+FLUTTER_EXPORT
+@protocol FlutterMessageCodec
```

Comment:
I'll look into it. I first dismissed NDCoder/NSCoding as too big a gun for this simple situation. Please be aware that plugin writers are not exposed to the internals of the Flutter standard codec, except they are given a list of supported values as part of the documentation.